examples/tutorials/date_time_charts.py

```diff
@@ -0,0 +1,351 @@
+# (bottom) border, and removing grid lines
+# - Using circles to plot data points defined through ``c``
+# in ``style`` parameter
+
+x = [
+    datetime.datetime(2021, 1, 1, 3, 45, 1),
+    datetime.datetime(2021, 1, 1, 6, 15, 1),
+    datetime.datetime(2021, 1, 1, 13, 30, 1),
+    datetime.datetime(2021, 1, 1, 20, 30, 1),
+]
+y = [5, 3, 1, 2]
+
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/5c",
+    region=[
+        datetime.datetime(2021, 1, 1, 0, 0, 0),
+        datetime.datetime(2021, 1, 2, 0, 0, 0),
+        0,
+        6,
+    ],
+    frame=["WS", "af"],
+    x=x,
+    y=y,
+    style="c0.4c",
+    pen="1p",
+    color="blue",
+)
+fig.show()
+
+########################################################################################
+# Using ISO Format
+# ----------------
+#
+# In addition to Python's ``datetime`` library, PyGMT also supports passing times
+# in ISO format. Basic ISO strings are formatted as ``YYYY-MM-DD``
+# with each ``-`` delineated section marking the four digit year value, two digit
+# month value, and two digit day value respectively.
+#
+# When including time of day into ISO strings, the ``T`` character is used, as
+# can be seen in the following example. This character is immediately followed
+# by a string formatted as ``hh:mm:ss`` where each ``:`` delineated section marking
+# the two digit hour value, two digit minute value, and two digit second value
+# respectively. The figure in the following example is plotted over a horizontal
+# range of one year from 1/1/2016 to 1/1/2017.
+
+x = ["2016-02-01", "2016-06-04T14", "2016-10-04T00:00:15", "2016-12-01T05:00:15"]
+y = [1, 3, 5, 2]
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/5c",
+    region=["2016-01-01", "2017-01-1", 0, 6],
+    frame=["WSen", "afg"],
+    x=x,
+    y=y,
+    style="a0.45c",
+    pen="1p",
+    color="dodgerblue",
+)
+fig.show()
+
+###############################################################################
+# Mixing and matching datetime and ISO
+# ------------------------------------
+#
+# The following example provides context on how both ``datetime`` and ``ISO``
+# date data can be plotted using PyGMT. This can be helpful when dates and times
+# are coming from different sources, meaning conversions do not need to take place
+# between ISO and datetime in order to create valid plots.
+
+x = ["2020-02-01", "2020-06-04", "2020-10-04", datetime.datetime(2021, 1, 15)]
+y = [1.3, 2.2, 4.1, 3]
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/5c",
+    region=[datetime.datetime(2020, 1, 1), datetime.datetime(2021, 3, 1), 0, 6],
+    frame=["WSen", "afg"],
+    x=x,
+    y=y,
+    style="i0.4c",
+    pen="1p",
+    color="yellow",
+)
+fig.show()
+
+########################################################################################
+# Using :meth:`pandas.date_range`
+# -------------------------------
+#
+# In the following example, a ``pandas.date_range`` object is used to pass date
```

Comment:
`pandas.date_range` does not use cross-reference?

Reply:
`:meth:`pandas.date_range`` is currently not working properly. I'd appreciate some help on how to do the cross reference correctly. Thanks!